web/components/templates/sessions/sessionId/Tree/TreeView.tsx

```diff
@@ -292,83 +447,87 @@ const TreeView: React.FC<TreeViewProps> = ({
     table.toggleAllRowsExpanded();
   };
 
+  useEffect(() => {
+    setPathColor(treeData, setColors, colors);
+  }, [treeData]);
```

Comment:
i think if we move

    <ColorContext.Provider value={{ colors, setColors }}>
    
we can reduce some complexity here